test/src/dfu_test.cpp

```diff
@@ -0,0 +1,1154 @@
+#include "gtest/gtest.h"
```

Comment:
This is mostly a copy past from bm_protocol. It includes typedef/enum/function style differences and then removed any custom config/nvm partition/reset function specific and fakes and replaced them with the dfu wrapper code. Here is a link to the old test for comparison: https://github.com/bristlemouth/bm_protocol/blob/experiment/bring_over_service_files/test/src/common/bcmp_dfu_ut.cpp